www/src/pages/table/forms/TransformationForm.tsx

```diff
@@ -0,0 +1,289 @@
+import { WithStyles, withStyles } from "@material-ui/styles";
+import {
+  Button,
+  Radio,
+  Tag,
+  Space,
+  Table,
+  Modal,
+  Form,
+  Col,
+  Row,
+  Select,
+  InputNumber,
+} from "antd";
+import { observer } from "mobx-react";
+import { ActionType } from "@ant-design/pro-table";
+import { useRef, useState } from "react";
+import Editor from "@monaco-editor/react";
+import {
+  TransformationResult,
+  useStores,
+  Transformation,
+  Table as TableModel,
+} from "../../../models";
+const styles = {};
+
+export interface TransformationFormProps {
+  type: string;
```

Comment:
I think you don't use this variable, am I right?

Reply:
type is used to open the form form menubar

Reply:
the field type is defined in the menubar, and you don't use it in this form.

Reply:
see the switch case in index.ts of forms

Reply:
updated the type